sql: investigate why catalog Lease tests do not work with a shared process virtual cluster

**Describe the problem**

When running some lease tests (all marked with this issue) with a shared process virtual cluster:
```
=== RUN   TestLeaseManagerReacquire
    test_log_scope.go:170: test logs captured to: /var/folders/sh/9k10t77143q0f1ffxxv99lbh0000gq/T/logTestLeaseManagerReacquire3104296236
    test_log_scope.go:81: use -show-logs to present logs inline
    lease_test.go:415: expected different leases, but found 104("foo") ver=1:1698146574.830708472,0, refcount=2
    panic.go:522: -- test log scope end --
test logs left over in: /var/folders/sh/9k10t77143q0f1ffxxv99lbh0000gq/T/logTestLeaseManagerReacquire3104296236
--- FAIL: TestLeaseManagerReacquire (3.33s)
```

**How to reproduce**

Set the following in `TestServerArgs` to force a shared process tenant and cause the failure to happen.
```
DefaultTestTenant: base.SharedTestTenantAlwaysEnabled,
```

**Expected behavior**
The test should pass when running with a shared process virtual cluster, unless after investigation the test is expected not to work with one.

Ref: https://github.com/cockroachdb/cockroach/issues/112857
Epic CRDB-48944
Jira issue: CRDB-32694

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -2181,9 +2181,6 @@ func TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces(t *testing.T) {
 	blockLeaseAcquisitionOfInterestingTable := make(chan chan struct{})
 	unblockAll := make(chan struct{})
 	args := base.TestServerArgs{}
-	args.DefaultTestTenant = base.TestDoesNotWorkWithSharedProcessModeButWeDontKnowWhyYet(
-		base.TestTenantProbabilistic, 112957,
-	)
 	tc := testcluster.StartTestCluster(t, 1, base.TestClusterArgs{
 		ServerArgs: args,
 	})
@@ -2818,9 +2815,6 @@ func TestLeaseTxnDeadlineExtensionWithSession(t *testing.T) {
 	var txnID string
 
 	var params base.TestServerArgs
-	params.DefaultTestTenant = base.TestDoesNotWorkWithSharedProcessModeButWeDontKnowWhyYet(
-		base.TestTenantProbabilistic, 112957,
-	)
 	params.Settings = cluster.MakeTestingClusterSettings()
 	// Set the lease duration such that the next lease acquisition will
 	// require the lease to be reacquired.

